### PR TITLE
linting: added errcheck, gocritic, goimports, golint, govet & megacheck

### PR DESCRIPTION
**Pull request by lopezator**
added linters using golangci. fix linted errors.

references #47

### CODE REVIEW

**Comment by lopezator** [comment]
Just a first pass.

Needs revision, maybe adjustements, and a explanation of how to install golangci-lint. 

Maybe add instructions to README?

Integration of sanity-check target with CI?

CC @F21

**Comment by lopezator** [comment]
PTAL @F21 

Feel free to suggest/change anything. English is not my native language and the description in the README.md file for example, might not be clear enough.

I left the Makefile although it is calling only `golangci-lint run` in case we need to add more checks to the `sanity-check` target, like `go mod verify` or whatever without having the need to change the wercker.yml file.

Cannot make the build pass, looks something related to go modules, not sure.

**Comment by F21** [comment]
@lopezator Thanks again for working on this! There are a few minor changes I'd like for the README.md file, but let's see if we can fix the go modules error first.

I did a quick search and it appears that `golangci-lint` supports go modules: https://github.com/golangci/golangci-lint/issues/205

I believe what we need to do is add `golangci-lint` to our go.mod. See https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module

In essence, you'll need to create a `tools.go` file and import `golangci-lint` with an underscore and add a build constraint `// +build tools`. Then you need to `go get` `golangci-lint` to add it to `go.mod`.

In the wercker.yml file, rather than installing the `golangci-lint` binary, we use `go install` to install it. This should install the correct version of `golangci-lint` as specified in our `go.mod` file. 

I haven't had  the chance to give this a shot, but I think it should work.

**Comment by lopezator** [comment]
after messing around with this wercker thing for a while it seems I've been able to pass the checks. PTAL @F21 let's see if we can polish this :)

**Comment by F21** [comment]
@lopezator Thanks for your work on this! The problem with using `base-path` is that the project is loaded into the `GOPATH` and `GO111MODULE` would be set to `auto`. In general, I prefer to keep the code out of the `GOPATH` when using go modules.

I have pushed a commit that does the following:
1. Pin the `golangci-lint` version in `go.mod` and added it to `tools.go`. This way, running `go install github.com/golangci/golangci-lint/cmd/golangci-lint` will always install the same version.
2. Regenerated `go.sum` because it was generated using an older version of `Go 1.11` and the checksum algorithm changed between patch releases (causing mismatch errors).
3. Made some adjustments to the README.

The commit is on the `golint` branch in this repo: https://github.com/Boostport/migration/commit/44263f47d050f1c32abd4a839c6538a71e86e0fb

The CI  for my fork also passes except for code climate (due to some missing config): https://app.wercker.com/F21/migration/runs/build/5c37cb980f540f0025857fb8?step=5c37cbc2cfa0fc000779b757

Can you squash my commit with yours into 1 commit (you should be set as the author of the commit)? I think we should be ready to merge 👍

**Comment by lopezator** [comment]
I've cherry-picked that commit and squashed it into mine, thanks for the help @F21 

PTAL again 🙂

File: README.md [hunk 1]
```diff
@@ -207,5 +207,17 @@ Phoenix driver, which uses the scheme to determine if we should connect over `ht
 project is structured, it was also almost impossible to add support for embeddable migration files without major
 changes.
 
+## Contributing
+We run automatically some linters using [golangci-lint](https://github.com/golangci/golangci-lint) to check code quality
```
Comment (F21) [inline README.md]: Looks like I messed up a bit here. Can you change it to `We automatically run`?

**Comment by F21** [comment]
Many thanks for pulling in the changes. I messed up a bit with the README, do you mind making the change?

**Comment by F21** [comment]
Thanks @lopezator !

**Pull request merged**